modules/Microsoft.App/managedEnvironments/deploy.bicep

```diff
@@ -0,0 +1,171 @@
+}
+
+resource managedEnvironment 'Microsoft.App/managedEnvironments@2022-06-01-preview' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: skuName
+  }
+  properties: {
+    appLogsConfiguration: {
+      destination: logsDestination
+      logAnalyticsConfiguration: {
+        customerId: logAnalyticsWorkspace.properties.customerId
+        sharedKey: logAnalyticsWorkspace.listKeys().primarySharedKey
+      }
+    }
+    daprAIConnectionString: daprAIConnectionString
+    daprAIInstrumentationKey: daprAIInstrumentationKey
+    customDomainConfiguration: {
+      certificatePassword: certificatePassword
+      certificateValue: !empty(certificateValue) ? certificateValue : null
+      dnsSuffix: dnsSuffix
+    }
+    vnetConfiguration: {
+      dockerBridgeCidr: dockerBridgeCidr
+      infrastructureSubnetId: infrastructureSubnetId
+      internal: internal
+      outboundSettings: !empty(vnetOutboundSettings) ? vnetOutboundSettings : null
+      platformReservedCidr: platformReservedCidr
+      platformReservedDnsIP: platformReservedDnsIP
+      runtimeSubnetId: runtimeSubnetId
+    }
+    workloadProfiles: !empty(workloadProfiles) ? workloadProfiles : null
+    zoneRedundant: zoneRedundant
+  }
+}
+
+module managedEnvironment_roleAssignments '.bicep/nested_roleAssignments.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${uniqueString(deployment().name, location)}-ManagedEnvironment-Rbac-${index}'
+  params: {
+    description: contains(roleAssignment, 'description') ? roleAssignment.description : ''
+    principalIds: roleAssignment.principalIds
+    principalType: contains(roleAssignment, 'principalType') ? roleAssignment.principalType : ''
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    condition: contains(roleAssignment, 'condition') ? roleAssignment.condition : ''
+    delegatedManagedIdentityResourceId: contains(roleAssignment, 'delegatedManagedIdentityResourceId') ? roleAssignment.delegatedManagedIdentityResourceId : ''
+    resourceId: managedEnvironment.id
+  }
+}]
+
+resource managedEnvironment_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock)) {
+  name: '${managedEnvironment.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: managedEnvironment
+}
+
+@description('The name of the resource group the Container Apps Managed Environment was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The location the resource was deployed into.')
+output location string = managedEnvironment.location
+
+@description('Managed Envrionment Name.')
+output name string = managedEnvironment.name
+
+@description('Managed environment ID.')
+output resourceId string = managedEnvironment.id
+
+@description('The name of the Log analytics workspace name.')
```

Comment:
Is there a specific reason we are outputting the LAW name, as its already an input parameter as resource ID

Reply:
No, I removed it in a few places and apparently missed one

Reply:
done